### ISSUE TITLE
workspace/didChangeWatchedFiles error from to extensive globPattern

### ISSUE THREAD

**Issue by jonaskrask** [jonaskrask/vdm-vscode#52]
The "workspace/didChangeWatchedFiles" reports changes to files with "git:" uri prefix, these should not be watched as far as I understand from https://github.com/microsoft/vscode/issues/91242.

From the linked issue it seems we need to specify "file:" in the globPattern in LSPWorkspaceManager.java (I think this is where the file watcher is setup, correct me if I'm wrong)

**Log message of the error**
11:22:53.731: >>> LSP {"jsonrpc":"2.0","method":"workspace/didChangeWatchedFiles","params":{"changes":[{"uri":"git:/c%3A/Users/au543131/Documents/Github/vdm-vscode/vdm_test_files/Alarm%2B%2BtracesPP/test1.vdmpp.git?%7B%22path%22%3A%22c%3A%5C%5CUsers%5C%5Cau543131%5C%5CDocuments%5C%5CGithub%5C%5Cvdm-vscode%5C%5Cvdm_test_files%5C%5CAlarm%2B%2BtracesPP%5C%5Ctest1.vdmpp%22%2C%22ref%22%3A%22%22%7D","type":2}]}}
11:22:53.731: EXCEPTION: IllegalArgumentException URI scheme is not "file"
11:22:53.732: <<< LSP { "jsonrpc" : "2.0", "error" : { "code" : -32603, "message" : "URI scheme is not \"file\"" }, "id" : null }

**Comment by nickbattle** [comment]
I think the problem here is partly that we have to specify "*" so that we are notified of new folder creations. If we just say "*.vdm??" or similar, then creating a new folder does not update the Server and those files are ignored until a restart.

How was that "git:" URI created? If I can recreate the message, I can probably add some "ignore this" defence.

**Comment by jonaskrask** [comment]
Yes, that should probably also be included. But from what I gather from https://github.com/microsoft/vscode/issues/91242 there could potentially also be other prefixes/file systems that can occur than `uri:`. I would think that the be way to avoid this is to not watch for changes that we don't want to react on.

I am not entirely sure what caused the event, but it happens while I have the folder https://github.com/jonaskrask/vdm-vscode/tree/master/vdm_test_files/Alarm%2B%2BtracesPP open in VSCode (cloned) and GitKraken open on the vdm-vscode repository. It doesn't seem to happen when GitKraken is not open.

**Comment by nickbattle** [comment]
Ah! I didn't appreciate that you can add URI schemes to the pattern. In that case, switching to `file:/**` or similar should fix this. I'll try it :)

**Comment by nickbattle** [comment]
Mmm... but that doesn't work. If I change it to use a `file://**` pattern, directory creations are not notified. The confuses it a bit. With just `/**` you get:
```
14:07:08.059: >>> LSP {"jsonrpc":"2.0","method":"workspace/didChangeWatchedFiles","params":{"changes":[{"uri":"file:///home/nick/vscode.tests/TestSL/libs","type":1}]}}
14:07:08.059: New directory created: /home/nick/vscode.tests/TestSL/libs
```
Though note that this is a "file" URI... curious.

**Comment by jonaskrask** [comment]
Hmm weird, would `file://**/*` be better?

**Comment by nickbattle** [comment]
No, just tried that. It's the same: it sends nothing about directory creation or deletion.

I can easily add a check in the handler for the didChangeWatchedFile to silently ignore anything that isn't a file: URI?

**Comment by nickbattle** [comment]
I've just pushed a tweak to ignore anything except file URIs.

This is fishy though. It would be interesting to see what registerOptions other language based extensions use for didChangeWatchedFiles. Presumably they need to identify new file and folder creations, just as we do. The apparent behaviour of the pattern matching is unintuitive.

**Comment by jonaskrask** [comment]
Okay, that will work for now, but yes we should check what other do. I can take a look at it next week
